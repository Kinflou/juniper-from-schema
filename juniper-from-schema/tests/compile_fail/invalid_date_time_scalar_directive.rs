#![allow(dead_code, unused_variables, unused_must_use, unused_imports)]
include!("../compile_pass/setup.rs");

use chrono::prelude::*;

juniper_from_schema::graphql_schema! {
    schema {
        query: Query
    }

    type Query {
        dateTime: DateTimeUtc! @juniper(ownership: "owned")
    }

    scalar DateTimeUtc @juniper(with_time_zone: "foobar")
}

pub struct Query;

impl QueryFields for Query {
    fn field_date_time(&self, _: &Executor<Context>) -> FieldResult<NaiveDateTime> {
        unimplemented!()
    }
}
