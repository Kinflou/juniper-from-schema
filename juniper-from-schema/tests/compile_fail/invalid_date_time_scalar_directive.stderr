error: proc macro panicked
  --> $DIR/invalid_date_time_scalar_directive.rs:6:1
   |
6  | / juniper_from_schema::graphql_schema! {
7  | |     schema {
8  | |         query: Query
9  | |     }
...  |
15 | |     scalar DateTimeUtc @juniper(with_time_zone: "foobar")
16 | | }
   | |_^
   |
   = help: message:

           error: Unsupported directive.
            --> schema:3:1
             |
           3 |    @ juniper(with_time_zone : "foobar")
             |    ^

           Invalid type. Expected `Boolean`, got `String`

           aborting due to previous error

error[E0405]: cannot find trait `QueryFields` in this scope
  --> $DIR/invalid_date_time_scalar_directive.rs:20:6
   |
20 | impl QueryFields for Query {
   |      ^^^^^^^^^^^ not found in this scope
